src/App.tsx

```diff
@@ -17,18 +65,32 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onChangeFormState={updateFormState}
+                formState={filterState}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+              {!loading && todos.length > 0 && (
+                <TodoList
+                  todos={visibleTodos}
+                  selectedTodoId={selectedTodoId}
+                  onSelectTodoId={(id) => setSelectedTodoId(id)}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          todo={selectedTodo}
+          onClose={(id) => setSelectedTodoId(id)}
```

Comment:
Have we ever set id on close? As I understand we are setting only null, so let's create a separate function for closing modal and set there selectedTodo to null
```suggestion
          onClose={handleCloseTodoModal}
```